File: app/src/main/java/com/app/id/dagger/AppComponent.java
package com.app.id.dagger;

import com.app.id.application.Application;
import com.app.id.dagger.modules.ActivityModule;
import com.app.id.dagger.modules.ApiModule;
import com.app.id.dagger.modules.AppModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, ApiModule.class})
public interface AppComponent {
    void inject(Application app);

    ActivityComponent plus(ActivityModule activityModule);
}

File: app/src/main/java/com/app/id/dagger/AppComponent.kt
package com.app.id.dagger

import com.app.id.application.Application
import com.app.id.dagger.modules.ActivityModule
import com.app.id.dagger.modules.ApiModule
import com.app.id.dagger.modules.AppModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class, ApiModule::class))
interface AppComponent {
    fun inject(app: Application)
    operator fun plus(activityModule: ActivityModule): ActivityComponent
}
